lib/std/setutils.nim

```diff
@@ -0,0 +1,32 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2020 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This module adds functionality to the builtin set
+## See also std/packedsets, set/sets
+
+import typetraits
+
+type SetElement* = char|byte|bool|int16|uint16|enum|uint8|int8
+  ## The allowed types of a builtin set.
```

Comment:
it's a bit more complicated, because set also allows transitive closure via `distinct` and `range`.

Should we simply remove (or disable with when false to deal with this in future PR) this type, since (as I argued below) you don't even need the
```
static: doAssert E is SetElement, $elementType(iter) & " does not yield a `SetElement`"
```
check?

Reply:
I guess I will remove the check, should I also remove the SetElement type aswell?

Reply:
maybe comment it out so it's "there" for future PR's